src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -527,12 +527,18 @@ class MacroAssembler: public Assembler {
   }
 
   // mv
-  void mv(Register Rd, address addr)          { li(Rd, (int64_t)addr); }
+  void mv(Register Rd, address addr)                  { li(Rd, (int64_t)addr); }
+  void mv(Register Rd, address addr, int32_t &offset) {
+    // Split address into a lower 12-bit sign-extended offset and the remainder,
+    // so that the offset could be encoded in jalr or load/store instruction.
+    offset = ((int32_t)(uintptr_t)addr << 20) >> 20;
```

Comment:
Would you mind one more tweak here? I think it will be more readable here if we convert 'addr' into int64_t type instead of uintptr_t.

Reply:
Of course. Also changed other uintptr_t usages to int64_t.